### DISCUSSION TITLE
Is there a way to protect a LIST from being reset when using the 'restart' button?

### DISCUSSION THREAD

**Discussion by LilCelestial** [inkle/ink#777] [Writing in ink] [answered]
Hello again!
I've been trying to figure this one out for a couple days and I've had no luck.
Does anyone have a way, or an idea, of how to stop a list from being reset upon restart? I have achievements that unlock for each ending the reader/player reaches and want everything else to reset when they click restart except for that. The achievements are a list numbered one to eighteen, that I adjust when the ending is read and refer to at the very end using a conditional statement.

I've tried creating my own 'restart' button at the end of my file and rewriting all the variables before the button is pressed, but I can't reset knots seen, which a lot of my choices are based on. Unless, there is a way to do that?

**Comment by Selsynn** [comment, marked as answer]
The way I'm always doing it, it's treating the part I want to keep as an exception to the reset. 

I particularly like the LIST keyword for that (and I'll call it `Knowledge` or some other relevant name)
Each time the user is passing through something I want to keep between reset, I'll change the Knowledge list value. 

When doing the "reset" loop action, I'll first copy the current value of the Knowledge list value. Reinit ink state to the beginning, and erasing the Knowledge list var to the value I stocked just before. And keep going. 

So instead of just using the `*` and `+` to keep track if that's the first time, I will also do {Knowledge?DragonsBreath} to know if the player already went there at least once in another run.

For example, in js, this is the code I use in the part where the `#RESTART` tag is used
```js
if( tag == "RESTART" ) {
  var knowKeep = story.variablesState.$("knowKeep");
  restart();
  story.variablesState.$("knowKeep", knowKeep);
  return;
}
```

I hope this can help you.

**Reply by LilCelestial** [reply]
Maelle you are a lifesaver! Thank you for sharing your code. That placement didn't work for me for some reason, I had to place it in the 'function restart' section for it to work. Thanks again!

`function restart() {
        var achievements = story.variablesState.$("achievements");
        story.ResetState();
        story.variablesState.$("achievements", achievements);

        setVisible(".header", true);

        // set save point to here
        savePoint = story.state.toJson();

        continueStory(true);

        outerScrollContainer.scrollTo(0, 0);
    }`

**Reply by LilCelestial** [reply]
Don't why I can't get that code to sit right in the comment!

**Reply by Selsynn** [reply]
Yeah, it seems the code did change a bit since I use that trick and it may be handled differently now. 
The part important is to stock the value before the reset, and adding it back to it just after.

For the presentation; one tick is enough if you have text you want to put in focus, three is when you want to do more than one line. (three at the beginning and at the end)

Happy you could resolve your problem!
